### PR TITLE
click handling and removal of ts-loader

### PR DESCRIPTION
**Pull request by afshin**
WIP, we might need to rethink hover and mouse events in general.

### CODE REVIEW

File: example/webpack.conf.js [hunk 1]
```diff
@@ -1,15 +1,14 @@
 module.exports = {
-  entry: './index.ts',
+  entry: './build/index.js',
   output: {
     filename: './build/bundle.js'
   },
   resolve: {
-    extensions: ['', '.ts', '.js']
+    extensions: ['', '.js']
```
Comment (blink1073) [inline example/webpack.conf.js]: Is this still needed?

Comment (blink1073) [inline example/webpack.conf.js]: Defaults:  `["", ".webpack.js", ".web.js", ".js"]`

File: src/index.css [hunk 2]
```diff
@@ -41,6 +41,16 @@
   flex: 1 1 auto;
 }
 
+.p-CommandPalette-itemCaption {
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+}
+
+.p-CommandPalette-itemCaption:empty:before {
```
Comment (sccolbert) [inline src/index.css]: I don't think we should force items to be the same size:
capture

File: src/palette.ts [hunk 3]
```diff
@@ -342,17 +349,21 @@ class CommandPalette extends Widget {
...
     let ctor = this.constructor as typeof CommandPalette;
 
     //
-    for (let { title, items } of result) {
-      fragment.appendChild(ctor.createHeaderNode(title));
-      for (let item of items) {
-        fragment.appendChild(ctor.createItemNode(item));
-      }
-    }
+    result.forEach((section, i) => {
```
Comment (sccolbert) [inline src/palette.ts]: use an index based for-loop here. This is allocating a closure on each iteration.

File: src/palette.ts [hunk 4]
```diff
@@ -342,17 +349,21 @@ class CommandPalette extends Widget {
...
+      let title = section.title;
+      let category = section.category;
+      fragment.appendChild(ctor.createHeaderNode(title, category));
+      section.items.forEach((item, j) => {
+        fragment.appendChild(ctor.createItemNode(item, [i, j].join('-')));
```
Comment (sccolbert) [inline src/palette.ts]: use string substition: `${i-j}` instead of allocation a temporary array.

File: src/palette.ts [hunk 5]
```diff
@@ -362,10 +373,32 @@ class CommandPalette extends Widget {
...
+      !(isItem = target.hasAttribute('data-index'))) {
+        if (target === this.node) return;
+        target = target.parentElement;
+    }
+    if (isCategory) {
+      let category = target.getAttribute('data-category');
+      console.log('category search', category);
+    } else {
+      let indices = target.getAttribute('data-index');
+      let category = parseInt(indices.split('-')[0], 10);
```
Comment (sccolbert) [inline src/palette.ts]: `indices[0]`

File: example/webpack.conf.js (hunk 1 above)
Comment (afshin) [inline example/webpack.conf.js]: Oh cool, I didn't know that.

File: src/index.css (hunk 2 above)
Comment (afshin) [inline src/index.css]: Okay, cool. I can see a case for both, but if we're using this as our model, that makes sense. I'll remove it.

File: src/palette.ts (hunk 3 above)
Comment (afshin) [inline src/palette.ts]: Okay.

File: src/palette.ts (hunk 4 above)
Comment (afshin) [inline src/palette.ts]: Got it.

File: src/index.css [hunk 6]
```diff
@@ -42,6 +42,13 @@
 }
 
 
+.p-CommandPalette-itemCaption {
```
Comment (sccolbert) [inline src/index.css]: I don't think this rule should be the default. Long captions should probably wrap.

**Pull request merged**